recipe/set_bld_opts.bat

```diff
@@ -28,8 +28,9 @@ set BLD_OPTS=%WIN64% ^
     LIBDIR=%LIBRARY_LIB% ^
     INCDIR=%LIBRARY_INC% ^
     DATADIR=%LIBRARY_PREFIX%\share\gdal ^
+    DLLBUILD=1 ^
     BLOSC_CFLAGS="-I%LIBRARY_INC%" ^
-    BLOSC_LIBS=%LIBRARY_LIB%\libblosc.lib ^
+    BLOSC_LIBS=%LIBRARY_LIB%\blosc.lib ^
```

Comment:
This seems to be quite a breaking change for the ecosystem as a whole. We should either issue a repodata patch, or maybe change the name of the library back in blosc.

Reply:
Yep it's a shocker, I imagine it's broken a number of things (and it is so obscure that the solution isn't obvious). Renaming things to be different from how upstream has it is going to add more confusion, but happy to go with whatever you think.

Reply:
did upstream intentionally do this? I know they are ok with vendoring things so they likely see that as an acceptable change. however I'm trying to think if we should do some patching at conda-forge. References would be appreciated.

Reply:
Are both of these import libraries or is one a static library?

Reply:
It seems that `libblosc.lib` is a static lib and `blosc.lib` is an import lib. It used to be the other way around...

Note: I haven't actually got onto a Windows box to check this 100% but i'm pretty confident by the sizes of the libs and the fact our errors went away when I go the right one that this is the case...

Reply:
Thanks Sam! 🙏

Was that always the case and we had it backwards here or did something change in the recent build?

Reply:
It looks like it was the other way around in 1.21.0, order changed when 1.21.1 was built...

Reply:
Wonder if PR ( https://github.com/Blosc/c-blosc/pull/328 ) is related

Reply:
thats the only one that looked related from my quick glance as well.

Reply:
Sorry deleted my prior comment since the PR I was looking at came after the release. So even though it did look interesting, don't think that is the cause (unless we are including it as a patch)

The rest of these [CMake changes]( https://github.com/Blosc/c-blosc/commits/v1.21.1/CMakeLists.txt ) don't immediately look off to me, but that doesn't mean there isn't something subtle going on

Reply:
Worth noting we are removing the static libraries with PR ( https://github.com/conda-forge/blosc-feedstock/pull/56 )

Reply:
I think these are the relevant lines (in `blosc` subdir): https://github.com/Blosc/c-blosc/blob/master/blosc/CMakeLists.txt#L211 and https://github.com/Blosc/c-blosc/blob/master/blosc/CMakeLists.txt#L136. But they don't seem to have changed between releases....

Reply:
IIUC they both have the same name? If so, maybe CMake is doing some disambiguation afterwards to ensure they don't stomp on each other. Perhaps just dropping the static library improves the situation

Reply:
I don't \*think\* that libs have the `lib` prefix by default under MSVC so they shouldn't be the same (only static gets the `lib` prefix)...